File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int levels(TreeNode root){
        if(root==null) return 0;
        return 1 + Math.max(levels(root.left),levels(root.right));
    }
    public void bfs( TreeNode root, int n, List<Integer> temp){
        if(root==null) return;
        if(n==1){
            temp.add(root.val);
        }
        bfs(root.left, n-1, temp);
        bfs(root.right, n-1, temp);
    }
    public List<Integer> rightSideView(TreeNode root) {
        int level = levels(root);
        
        List<Integer> ans = new ArrayList<>();
        for(int i = 1; i<=level; i++){
            List<Integer> temp = new ArrayList<>();
            bfs(root, i, temp);
            int n = temp.size();
            ans.add(temp.get(n-1));
        }

        return ans;
    }
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int levels(TreeNode* root){
        if(root==NULL) return 0;
        return 1 + max(levels(root->left), levels(root->right));
    }
    void bfs(TreeNode* root, int n, vector<int>&v){
        if(root==NULL) return;
        if(n==1){
            v.push_back(root->val);
        }
        bfs(root->left, n-1, v);
        bfs(root->right, n-1, v);
    }
    vector<int> rightSideView(TreeNode* root) {
        int level = levels(root);

        vector<int>ans;
        for(int i =1 ;i<=level;i++){
            vector<int>v;
            bfs(root ,i,v);
            int n = v.size();
            ans.push_back(v[n-1]);
        }
       return ans;
    }
};